spec/models/accounting/bank_spec.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Accounting::Bank, type: :model do # rubocop:disable Metrics/BlockLength
+  context 'validations' do
+    it { should validate_presence_of(:name) }
+    it { should validate_presence_of(:country_code) }
+    it { should validate_presence_of(:routing_number) }
+    it { should validate_presence_of(:swift_code) }
+    it { should validate_presence_of(:website) }
+    # TODO: not sure why when turning the values into uppercase in the model
```

Comment:
Tests are breaking and this is the console error:
![matchers error](https://github.com/RodSoriano/cocoa/assets/95240720/29f0099b-89d0-4b76-bf3d-5f1ac02472f9)

You can see here that it saves a record first and then it tries to save another one to compare it, the thing is that in the model you have a `before_validation` that will run every time a validation is run and this is leading to unexpected behaviour and I think is because you are doing `self` in this method.

What I think its happening is that because you are assigning a value using `self` it is creating another copy of the class and in this copy it expects all the other fields, triggering the other validation messages since it is creating a copy at the same time so `rspec` just brings both exceptions, I remove the `self` and everything run correctly.

Here is the code:
![codigo](https://github.com/RodSoriano/cocoa/assets/95240720/f1cf5567-aabb-4258-b16f-9bade42eb0c5)

Here are the passing specs:
![specs](https://github.com/RodSoriano/cocoa/assets/95240720/a438528f-23f3-4700-9bc3-cb3bd2161b42)